Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -4386,173 +4456,18 @@ void RenderBox::computePositionedLogicalWidthUsing(SizeType widthType, Length lo
-                    marginLogicalLeftValue = 0;
-                    marginLogicalRightValue = availableSpace; // will be negative
-                } else {
-                    marginLogicalLeftValue = availableSpace; // will be negative
-                    marginLogicalRightValue = 0;
-                }
-            }
-        } else if (inlineConstraints.marginBefore.isAuto()) {
-            // Solve for left margin
-            marginLogicalRightValue = inlineConstraints.marginAfterValue();
-            marginLogicalLeftValue = availableSpace - marginLogicalRightValue;
-        } else if (inlineConstraints.marginAfter.isAuto()) {
-            // Solve for right margin
-            marginLogicalLeftValue = inlineConstraints.marginBeforeValue();
-            marginLogicalRightValue = availableSpace - marginLogicalLeftValue;
-        } else {
-            // Over-constrained, solve for left if direction is RTL
-            marginLogicalLeftValue = inlineConstraints.marginBeforeValue();
-            marginLogicalRightValue = inlineConstraints.marginAfterValue();
-
-            // Use the containing block's direction rather than the parent block's
-            // per CSS 2.1 reference test abspos-non-replaced-width-margin-000.
-            if (!inlineConstraints.isOrthogonal() && !inlineConstraints.isLogicalLeftInlineStart())
-                logicalLeftValue = (availableSpace + logicalLeftValue) - marginLogicalLeftValue - marginLogicalRightValue;
-        }
-    } else {
-        /*--------------------------------------------------------------------*\
-         * Otherwise, set 'auto' values for 'margin-left' and 'margin-right'
-         * to 0, and pick the one of the following six rules that applies.
-         *
-         * 1. 'left' and 'width' are 'auto' and 'right' is not 'auto', then the
-         *    width is shrink-to-fit. Then solve for 'left'
-         *
-         *              OMIT RULE 2 AS IT SHOULD NEVER BE HIT
-         * ------------------------------------------------------------------
-         * 2. 'left' and 'right' are 'auto' and 'width' is not 'auto', then if
-         *    the 'direction' property of the containing block is 'ltr' set
-         *    'left' to the static position, otherwise set 'right' to the
-         *    static position. Then solve for 'left' (if 'direction is 'rtl')
-         *    or 'right' (if 'direction' is 'ltr').
-         * ------------------------------------------------------------------
-         *
-         * 3. 'width' and 'right' are 'auto' and 'left' is not 'auto', then the
-         *    width is shrink-to-fit . Then solve for 'right'
-         * 4. 'left' is 'auto', 'width' and 'right' are not 'auto', then solve
-         *    for 'left'
-         * 5. 'width' is 'auto', 'left' and 'right' are not 'auto', then solve
-         *    for 'width'
-         * 6. 'right' is 'auto', 'left' and 'width' are not 'auto', then solve
-         *    for 'right'
```

Comment:
A complaint has been has been that there is other enough comments in the code so we should probably keep something like this around in some form, or at least have spec links.

Reply:
There's a spec reference in the `resolvePosition()` function. I can add another one to the sizing functions.